### PR TITLE
Make TestServer.make_url compatible with yarl.URL

### PR DESCRIPTION
**Pull request by imbolc**
## Are there changes in behavior for the user?

Nope

## Related issue number

https://github.com/KeepSafe/aiohttp/issues/1389

### CODE REVIEW

File: aiohttp/test_utils.py [hunk 1]
```diff
@@ -78,6 +78,8 @@ def _make_factory(self):
         pass
 
     def make_url(self, path):
+        if isinstance(path, URL):
```
Comment (asvetlov) [inline aiohttp/test_utils.py]: I believe `return self._root.join(URL(path))` is easier solution.

Comment (imbolc) [reply aiohttp/test_utils.py]: I'm not sure what `path.startswith('/')` is checking, but in your solution if we pass `http://foo.com/` as `path` it returns `http://foo.com/` for any `self._root`. Is it ok?

Comment (asvetlov) [reply aiohttp/test_utils.py]: I thinks yes, it's just testing tool.
`.startswith()` does test for relative URL.
The same could be done by `assert not URL(path).is_absolute()`.
Not sure if we need this check but it is harmless.

**Comment by imbolc** [comment]
Done

**Comment by asvetlov** [comment]
Thanks!

**Comment by asvetlov** [comment]
Will be published as part of aiohttp 1.1.4 soon.

**Pull request merged**